### PR TITLE
refactor legacy facts to add compatibility with puppet 8

### LINKED ISSUE #12: deprecate legacy facts
With puppet 8 released, legacy facts have been deprecated by default, `$::operatingsystemrelease` should be replaced with `$facts['os']['release']['full']`. I can submit an MR, but I am not sure how to properly modify spec tests
```
    let :facts do
      {:operatingsystemrelease => '6.10'}
    end
```

### PR DESCRIPTION
**Pull request by vchepkov**

### CODE REVIEW

File: metadata.json [hunk 1]
```diff
@@ -22,7 +22,7 @@
   "requirements": [
     {
       "name": "puppet",
-      "version_requirement": ">= 6.1.0 < 8.0.0"
+      "version_requirement": ">= 6.1.0 < 9.0.0"
```
Comment (ekohl) [inline metadata.json]: I'd be hesitant to add this until it's actually available. I don't see it in on https://yum.puppet.com/ just yet.

**Review by ekohl — COMMENTED** [review]
One minor comment.

On a related note I wonder if we still need to keep the old support. https://access.redhat.com/articles/4038291 states that RHEL 7.4 EUS went EOL in 2019. I doesn't list 6.9 specifically, but I think that means 6.8 and 6.9 were never EUS.

@lphiri is it safe to drop the old style by now?

File: metadata.json (hunk 1 above)
Comment (vchepkov) [reply metadata.json]: that's fine, I will revert this

**Review by ekohl — APPROVED** [review]
I'll note that technically Puppet 7.21.0 added an option to disable legacy facts, but only version 8 defaults that to off.

**Pull request merged**